ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/CommandTransactionChecks.scala

```diff
@@ -423,6 +426,98 @@ abstract class CommandTransactionChecks
         }
       }
 
+      "permit fetching a divulged contract" in forAllMatchingFixtures {
+        case TestFixture(SandboxInMemory, ctx) =>
+        def pf(label: String, party: String) =
+          RecordField(label, Some(Value(Value.Sum.Party(party))))
+        val odArgs = Seq(pf("owner", owner), pf("delegate", delegate))
+        val delegatedCreate = simpleCreate(
+          ctx,
+          cid("SDVl3"),
+          owner,
+          templateIds.delegated,
+          Record(Some(templateIds.delegated), Seq(pf("owner", owner))))
+        val delegationCreate = simpleCreate(
+          ctx,
+          cid("SDVl4"),
+          owner,
+          templateIds.delegation,
+          Record(Some(templateIds.delegation), odArgs))
+        val showIdCreate = simpleCreate(
+          ctx,
+          cid("SDVl5"),
+          owner,
+          templateIds.showDelegated,
+          Record(Some(templateIds.showDelegated), odArgs))
+        val exerciseOfFetch = for {
+          delegatedEv <- delegatedCreate
+          delegationEv <- delegationCreate
+          showIdEv <- showIdCreate
+          fetchArg = Record(
+            None,
+            Seq(RecordField("", Some(Value(Value.Sum.ContractId(delegatedEv.contractId))))))
+          showResult <- failingExercise(
```

Comment:
The test case is `permit fetching a divulged contract` why are we testing for failures here (`// Exercise a choice that is supposed to fail.`)?
Plus it is the same owner everywhere, should not it work? Maybe the scenario description should be changed... why it should be permitted or not permitted.

Reply:
We're testing that the result is `Code.OK`; see a few lines below.

Reply:
I mean where is the test that `delegate` can fetch? If it is being tested through `failingExercise` with `Code.OK`, it is confusing. Why do we call it `failingExercise` instead of `successfullExercise`?

this is the original owner says `ShowIt` and it is successfull, right? Because it is `Code.OK`
```
          showResult <- failingExercise(
            ctx,
            cid("SDVl6"),
            submitter = owner,
            template = templateIds.showDelegated,
            contractId = showIdEv.contractId,
            choice = "ShowIt",
            arg = Value(Value.Sum.Record(fetchArg)),
            Code.OK,
            pattern = ""
          )
```
and this is delegate fetches it:
```
          fetchResult <- failingExercise(
            ctx,
            cid("SDVl7"),
            submitter = delegate,
            template = templateIds.delegation,
            contractId = delegationEv.contractId,
            choice = "FetchDelegated",
            arg = Value(Value.Sum.Record(fetchArg)),
            Code.OK,
            pattern = ""
          )
```
and it is again successful because because `Code.OK` does not matter that the function called `failingExercise`

Reply:
Your understanding of what's going on is perfectly cromulent.